sw/device/lib/dif/meson.build

```diff
@@ -0,0 +1,23 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# MMIO register manipulation library
+sw_dif_lib_register = declare_dependency(
+  link_with: static_library(
+    'sw_dif_lib_register',
+    sources: ['register.c'],
+  )
+)
+
+# SPI device DIF library
+sw_dif_spi_device = declare_dependency(
```

Comment:
The `declare_dependency` wrapper was used to associate the reg_h file with the library. You may be able to just use `sw_dif_spi_device = static_library(...)` now.

Reply:
You still need to write `declare_dependency()` to be able to depend on the resulting static library, which we'll wind up doing later on.

Reply:
yep, that is one option. Using `declare_dendency` allows you to use this as a dependency in other targets, and also bundle additional sources or headers with the dependency object.

With `static_library` you'd have to use the `link_with` option in other targets. 

I side with consistency, but just wanted to comment on this as an option in case we find `declare_dependency(link_with:static_library())` too verbose